### PR TITLE
CODENVY-1758; add OAuth2 authenticator for Linkedin

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Added OAuth2 authentication via LinkedIn. Configured via codenvy.env file with following variables:
```
CODENVY_LINKEDIN_CLIENT_ID=your_linkedin_client_id
CODENVY_LINKEDIN_SECRET=your_linkedin_secret
```

### What issues does this PR fix or reference?
#1758 

#### Changelog
Added oAuth2 authentication via LinkedIn.

#### Release Notes
We have added a new oAuth provider to Codenvy. You can now use your LinkedIn credentials to sign up or sign in.

#### Docs PR
https://github.com/codenvy/docs/pull/81

### CODE REVIEW

**Comment by skabashnyuk** [comment]
LinkedInOauthAuthenticator -> LinkedInOAuthAuthenticator

File: plugins/plugin-linkedin/codenvy-plugin-linkedin-oauth2/src/main/java/org/eclipse/che/security/oauth/LinkedInOauthAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    public User getUser(OAuthToken accessToken) throws OAuthAuthenticationException {
+        Map<String, String> params = new HashMap<>();
+        params.put("Authorization", "Bearer " + accessToken.getToken());
+        try {
+            // LinkedIn specific trick (default user URL does not returns email)
+            final String values = "(email-address,first-name,last-name)";
+            final String query = String.format("format=json");
+            JsonValue userValue = doRequest(new URL(userUri + ":" + values + "?" + query), params);
+            User user = new LinkedInUser();
+            user.setEmail(userValue.getElement("emailAddress").getStringValue());
```
Comment (skabashnyuk) [inline plugins/plugin-linkedin/codenvy-plugin-linkedin-oauth2/src/main/java/org/eclipse/che/security/oauth/LinkedInOauthAuthenticator.java]: Maybe you should check existance ```userValue.getElement("emailAddress")``` of this element?

**Comment by skabashnyuk** [comment]
@codenvy/pm please approve

**Comment by TylerJewell** [comment]
Release notes need to be revisited.  The changelog is ok but the release notes are meant for marketing.  So we should talk about how to configure it within configuration, are there any logos that appear, and whether there is a difference between onprem and saas.

**Comment by bmicklea** [comment]
To be clear:
@mshaposhnik:
- Will this be included in SaaS and On-Prem assemblies by default?
- Can it be turned on/off in configuration?
- Provide a screenshot of the new signup/login screen with the LinkedIn logo for oAuth

PM will take this information and update the release note section with it.

**Comment by mshaposhnik** [comment]
@bmicklea  1,2  - yes,  3 - 
linked_out

**Comment by JamesDrummond** [comment]
This will need docs. Not sure if I like the look and it doesn't include the other oauth links. I would think round corners like the google oauth icon below. WDYT @bmicklea .

clipboard

**Comment by bmicklea** [comment]
@mshaposhnik where are those variables found (and it looks like you listed the same one twice, are there two?)

I don't see them in the `codenvy.env` file.

**Comment by JamesDrummond** [comment]
@bmicklea Think @mshaposhnik meant `CODENVY_LINKEDIN_SECRET` for the second variable. I updated the description above a little while ago.

**Comment by bmicklea** [comment]
Thanks, I still don't see them in the `codenvy.env` file so where are they changed?

**Comment by skabashnyuk** [comment]
@bmicklea https://github.com/codenvy/codenvy/pull/1791/files#diff-3dd59f9815c904d76e88a331b6820074R195

**Comment by bmicklea** [comment]
Thanks Sergii.  I don't think we usually want people messing with the `codenvy.pp` so I'll skip that in the release notes. This should be on by default for Codenvy.io SaaS though.

**Comment by JamesDrummond** [comment]
@bmicklea You are looking at the wrong file on the link. The link https://github.com/codenvy/codenvy/pull/1791/files#diff-4e1211b38190902dee42a5833ea2efb5R466 highlights code for `codenvy.env` not `codenvy.pp` even though that is the filename you see at first when you first click the link. Line 466 of `codenvy.env` is highlighted.

**Pull request merged**